src/index.ts

```diff
@@ -82,16 +82,22 @@ function polyunionPass(
 
 			const targetFeature = fc.features[candidate.id];
 
-			if (
-				intersect(mergeCandidate, targetFeature) ||
-				booleanOverlap(mergeCandidate, targetFeature)
-			) {
-				mergeCandidate = union(mergeCandidate, targetFeature) as Feature<Polygon>;
-				processed.add(candidate.id);
+			// Attempt to merge the two features if they intersect or overlap
+			try {
+				if (
+					intersect(mergeCandidate, targetFeature) ||
+					booleanOverlap(mergeCandidate, targetFeature)
+				) {
+					mergeCandidate = union(mergeCandidate, targetFeature) as Feature<Polygon>;
+					processed.add(candidate.id);
```

Comment:
```suggestion
					const polygon = union(mergeCandidate, targetFeature) as Feature<Polygon>;
					if (polygon.geometry.type === 'Polygon') {
						mergeCandidate = polygon;
						processed.add(candidate.id);
					}
```

Reply:
I found out that when two polygons touch in a vertex, turf does not merge them. Instead, it returns a MultiPolygon.